src/multichain_demo.js

```diff
@@ -0,0 +1,1151 @@
+import * as d3 from "d3";
+import { Contract } from 'web3-eth-contract';
+import abi from './multichain_demo_contract_abi.json'
+const bridgeContract = new Contract(abi)
+
+//
+// Constants
+//
+const WeiPerEth = 1000000000000000000
+
+const BridgeableScopes = {
+    // Sepolia
+    "eip155:11155111": {
+        name: "Sepolia",
+        contractAddress: '0x2e2512fd69cba059DFf557cD6f683a3279402e91',
+        blockExplorerUrl: 'https://sepolia.etherscan.io',
+        supports: ["eip155:59141", "eip155:421614", "eip155:11155420", "eip155:168587773"]
+    },
+    // Linea Sepolia (no Arbitrium)
+    "eip155:59141": {
+        name: "Linea Sepolia",
+        contractAddress: '0x786Cb4C684F9D4bA4aBEbddE2c8a4D4ec80a9b78',
+        blockExplorerUrl: 'https://sepolia.lineascan.build/',
+        supports: ["eip155:11155111", "eip155:11155420", "eip155:168587773"]
+    },
+    // Arbitrum Sepolia (no Linea)
+    "eip155:421614": {
+        name: "Arbitrum Sepolia",
+        contractAddress: '0x697230B7c217F4F45C460d5d181F792AB0aC6549',
+        blockExplorerUrl: 'https://sepolia.arbiscan.io/',
+        supports: ["eip155:11155111", "eip155:11155420", "eip155:168587773"]
+    },
+    // OP Sepolia (no Blast)
+    "eip155:11155420": {
+        name: "OP Sepolia",
+        contractAddress: '0x697230B7c217F4F45C460d5d181F792AB0aC6549',
+        blockExplorerUrl: 'https://sepolia-optimism.etherscan.io/',
+        supports: ["eip155:11155111", "eip155:59141", "eip155:421614"]
+    },
+    // Blast Sepolia (no OP)
+    "eip155:168587773": {
+        name: "Blast Sepolia",
+        contractAddress: '0x697230B7c217F4F45C460d5d181F792AB0aC6549',
+        blockExplorerUrl: 'https://sepolia.blastscan.io/',
+        supports: ["eip155:11155111", "eip155:59141", "eip155:421614"]
+    },
+}
+
+const pastelColors = [
+    "#e8b6ae", "#f0c1ad", "#eed0ba", "#ebdec6", "#cdc0d6",
+    "#b8b1cf", "#959bad", "#97aeb8", "#a5bcc1", "#b2c9c9",
+    "#9ac7a8", "#b8dbc9", "#c5e3d2", "#d1ebdb", "#d6e1c8",
+    "#e5c28c", "#e8d097", "#edd9aa", "#f2e6bb", "#f9f3dd"
+];
+const NODE_MIN_RADIUS = 130;
+const NODE_PADDING = 20;
+const NODE_SCALE = 1.3;
+const NODE_TEXT_START = "-2.4em";
+const NODE_TEXT_SPACING = "1.2em";
+const NODE_FLASH_COLOR = "gray"
+const EDGE_COLOR = "#999";
+const EDGE_STROKE_WIDTH = 3;
+const FLASH_DURATION = 3000;
+
+const ETH_VALUE_PRECISION = 4;
+
+const USE_SUBSCRIPTIONS = true; // use false for polling
```

Comment:
set this to false to use polling instead of eth_subscribe to trigger on chain data fetching.

funnily enough, using polling results in a more responsive dapp experience than using subscription on block heads to determine when to read the chain for new data. I think this is because eth subscriptions causes us to hit the RPC rate limit, where as just using a setTimeout poll is close enough and results in less rate limiting